fortex/health/processors/scispacy_processor.py

```diff
@@ -0,0 +1,163 @@
+from scispacy.hyponym_detector import HyponymDetector
+
+from ftx.medical.clinical_ontology import Hyponym, Abbreviation, Phrase
+
+__all__ = [
+    "ScispaCyProcessor",
+]
+
+
+class ScispaCyProcessor(PackProcessor):
+    r"""
+    Implementation of this ScispaCyProcessor has been based on ScispaCy
+    pretained models (based on spacy),  A rendition of it that
+    exists on github has been referred to as well.
+
+    Referred repository link:
+    https://pythonlang.dev/repo/allenai-scispacy/
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.extractor = None
+
+    def set_up(self):
+        self.extractor = spacy.load(self.configs.model_name)
+        if self.configs.pipe_name == "abbreviation_detector":
+            self.extractor.add_pipe(self.configs.pipe_name)
+        else:  # hyponym
+            self.extractor.add_pipe(
+                self.configs.pipe_name, last=True, config={"extended": False}
+            )
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up()
+
+    def _process(self, input_pack: DataPack):
+        r"""
+        ScispaCyProcessor is done on the basis of
+        using SciSpacy and the corresponding
+        trained model for Hyponym, Abbreviation
+
+        """
+
+        path_str, module_str = self.configs.entry_type.rsplit(".", 1)
+        mod = importlib.import_module(path_str)
+        entry = getattr(mod, module_str)
+        # entry = get_class(class_name=module_str, module_paths=path_str)
+        for entry_specified in input_pack.get(entry_type=entry):
+
+            doc = self.extractor(entry_specified.text)
+
+            if self.configs.pipe_name == "abbreviation_detector":
+                list_of_abrvs = []
+                for abrv in doc._.abbreviations:
+                    tmp_abrv = Abbreviation(
+                        pack=input_pack, begin=abrv.start, end=abrv.end
+                    )
+                    tmp_abrv.long_form = abrv._.long_form
+                    list_of_abrvs.append(tmp_abrv)
+
+            else:
+                for item in doc._.hearst_patterns:
+                    general_concept: Phrase = Phrase(
+                        pack=input_pack, begin=item[1].start, end=item[1].end
+                    )
+                    specific_concept = Phrase(
+                        pack=input_pack, begin=item[2].start, end=item[2].end
+                    )
+                    hlink = Hyponym(
+                        pack=input_pack,
+                        parent=general_concept,
+                        child=specific_concept,
+                    )
+                    hlink.hyponym_link = item[0]
+
+    @classmethod
+    def default_configs(cls):
+        r"""
+        This defines a basic config structure for `ScispaCyProcessor`.
+
+        Following are the keys for this dictionary:
+         - `entry_type`: input entry type for classification
```

Comment:
documentation too brief and incorrect. 
1. Please make it clear for users are not familiar with our jargons. What is an entry? 
2. The purpose of this processor is also not `classification`.

Reply:
Corrected to let user to use the document type in the new comment.